pkg/workloadmeta/store.go

```diff
@@ -567,15 +620,52 @@ func notifyChannel(name string, ch chan EventBundle, events []Event, wait bool)
 	}
 }
 
+func classifyByKindAndID(entities []Entity) map[Kind]map[string]Entity {
+	res := make(map[Kind]map[string]Entity)
+
+	for _, entity := range entities {
+		kind := entity.GetID().Kind
+		entityID := entity.GetID().ID
+
+		_, found := res[kind]
+		if !found {
+			res[kind] = make(map[string]Entity)
+		}
+		res[kind][entityID] = entity
+	}
+
+	return res
+}
+
 // GetGlobalStore returns a global instance of the workloadmeta store,
 // creating one if it doesn't exist. Start() needs to be called before any data
 // collection happens.
 func GetGlobalStore() Store {
 	initOnce.Do(func() {
 		if globalStore == nil {
+			if remote {
+				collectorCatalog = map[string]collectorFactory{
+					"remote-workloadmeta": collectorCatalog["remote-workloadmeta"],
+				}
+			} else {
+				delete(collectorCatalog, "remote-workloadmeta")
+			}
+
 			globalStore = NewStore(collectorCatalog)
 		}
 	})
 
 	return globalStore
 }
+
+// ConfigureAsRemote configures workloadmeta to be remote. Returns an error if
+// the store has already been initialized.
+func ConfigureAsRemote() error {
+	if globalStore != nil {
+		return fmt.Errorf("cannot configure workloadmeta as remote because it has already been initialized")
+	}
+
+	remote = true
+
+	return nil
+}
```

Comment:
This part might change after https://github.com/DataDog/datadog-agent/pull/13830 is merged